packages/components/src/flex/flex/hook.js

```diff
@@ -78,34 +78,43 @@ export function useFlex( props ) {
 			marginBottom: wrap ? `calc(${ space( gap ) } * -1)` : undefined,
 		} );
 
-		sx.Items = css( {
-			/**
-			 * Workaround to optimize DOM rendering.
-			 * We'll enhance alignment with naive parent flex assumptions.
-			 *
-			 * Trade-off:
-			 * Far less DOM less. However, UI rendering is not as reliable.
-			 */
-			'> * + *:not(marquee)': {
-				marginTop: isColumn ? space( gap ) : undefined,
-				marginRight: ! isColumn && isReverse ? space( gap ) : undefined,
-				marginLeft:
-					! isColumn && ! isReverse ? space( gap ) : undefined,
-			},
-		} );
+		/**
+		 * Workaround to optimize DOM rendering.
+		 * We'll enhance alignment with naive parent flex assumptions.
+		 *
+		 * Trade-off:
+		 * Far less DOM less. However, UI rendering is not as reliable.
+		 */
+		sx.Items = css`
+			> * + *:not( marquee ) {
+				margin-top: ${ isColumn ? space( gap ) : undefined };
+				${ rtl( {
+					marginLeft:
+						! isColumn && ! isReverse ? space( gap ) : undefined,
+					marginRight:
+						! isColumn && isReverse ? space( gap ) : undefined,
+				} )() }
+			}
+		`;
 
-		sx.WrapItems = css( {
-			'> *:not(marquee)': {
-				marginBottom: space( gap ),
-				marginLeft: ! isColumn && isReverse ? space( gap ) : undefined,
-				marginRight:
-					! isColumn && ! isReverse ? space( gap ) : undefined,
-			},
-			'> *:last-child:not(marquee)': {
-				marginLeft: ! isColumn && isReverse ? 0 : undefined,
-				marginRight: ! isColumn && ! isReverse ? 0 : undefined,
-			},
-		} );
+		sx.WrapItems = css`
+			> *:not( marquee ) {
+				margin-bottom: ${ space( gap ) };
+				${ rtl( {
+					marginLeft:
+						! isColumn && isReverse ? space( gap ) : undefined,
+					marginRight:
+						! isColumn && ! isReverse ? space( gap ) : undefined,
+				} )() }
+			}
+
+			> *:last-child:not( marquee ) {
+				${ rtl( {
+					marginLeft: ! isColumn && isReverse ? 0 : undefined,
+					marginRight: ! isColumn && ! isReverse ? 0 : undefined,
+				} )() }
```

Comment:
```suggestion
				} ) }
```

See https://github.com/WordPress/gutenberg/pull/33729/files#r682696216